packages/invoices/src/OutboundShipment/api/api.ts

```diff
@@ -130,13 +138,36 @@ export const OutboundApi = {
   updateLines:
     (api: OutboundShipmentApi, storeId: string) =>
     async (draftStocktakeLines: DraftOutboundLine[]) => {
+      const filtered = draftStocktakeLines.filter(
+        ({ numberOfPacks }) => numberOfPacks > 0
+      );
       const input = {
-        insertOutboundShipmentLines: draftStocktakeLines
-          .filter(({ isCreated }) => isCreated)
-          .map(createInsertOutboundLineInput),
-        updateOutboundShipmentLines: draftStocktakeLines
-          .filter(({ isCreated, isUpdated }) => !isCreated && isUpdated)
-          .map(createUpdateOutboundLineInput),
+        insertOutboundShipmentLines: filtered
+          .filter(
+            ({ type, isCreated }) =>
+              isCreated && type === InvoiceLineNodeType.StockOut
+          )
+          .map(outboundParsers.toInsertLine),
+        updateOutboundShipmentLines: filtered
+          .filter(
+            ({ type, isCreated, isUpdated }) =>
+              !isCreated && isUpdated && type === InvoiceLineNodeType.StockOut
+          )
+          .map(outboundParsers.toUpdateLine),
+        insertOutboundShipmentUnallocatedLines: filtered
+          .filter(
+            ({ type, isCreated }) =>
+              type === InvoiceLineNodeType.UnallocatedStock && isCreated
+          )
+          .map(outboundParsers.toInsertPlaceholder),
+        updateOutboundShipmentUnallocatedLines: filtered
+          .filter(
+            ({ type, isCreated, isUpdated }) =>
+              type === InvoiceLineNodeType.UnallocatedStock &&
+              !isCreated &&
+              isUpdated
+          )
+          .map(outboundParsers.toUpdatePlaceholder),
```

Comment:
this is the main real change - adding the two fields `insertOutboundShipmentUnallocatedLines` & `updateOutboundShipmentUnallocatedLines`